include/aws/s3/private/s3_util.h

```diff
@@ -108,11 +111,38 @@ AWS_S3_API
 void replace_quote_entities(struct aws_allocator *allocator, struct aws_string *str, struct aws_byte_buf *out_buf);
 
 /* TODO could be moved to aws-c-common. */
+AWS_S3_API
 int aws_last_error_or_unknown(void);
 
 AWS_S3_API
 void aws_s3_add_user_agent_header(struct aws_allocator *allocator, struct aws_http_message *message);
 
+AWS_S3_API
+int aws_s3_parse_content_range_response_header(
+    struct aws_allocator *allocator,
+    struct aws_http_headers *response_headers,
+    uint64_t *out_range_start,
+    uint64_t *out_range_end,
+    uint64_t *out_object_size);
+
+AWS_S3_API
+int aws_s3_parse_content_length_response_header(
+    struct aws_allocator *allocator,
+    struct aws_http_headers *response_headers,
+    uint64_t *out_content_length);
+
+AWS_S3_API
+uint32_t aws_s3_get_num_parts(size_t part_size, uint64_t object_range_start, uint64_t object_range_end);
+
+AWS_S3_API
+void aws_s3_get_part_range(
```

Comment:
Same here, what are those params? It'll help if we can clarify it a little more than just naming.

Reply:
Added a more detailed comment.  Let me know if those params are still confusing!